app/models/foreman_openscap/policy.rb

```diff
@@ -67,8 +67,9 @@ def hostgroup_ids
 
     def hostgroup_ids=(ids)
       hostgroup_assets = []
-      ids.reject(&:empty?).map do |id|
-        hostgroup_assets << assets.where(:assetable_type => 'Hostgroup', :assetable_id => id).first_or_create!
+      ids.reject { |id| id.respond_to?(:empty?) && id.empty? }.map do |id|
+        params = { :assetable_type => 'Hostgroup', :assetable_id => id }
+        hostgroup_assets << (assets.where(params).first || Asset.new(params))
```

Comment:
Cool. Alas, it still bothers me a bit :)
Would you mind testing it with `.first_or_initialize` ? (sorry for the hustle)